pages/beckman_cherrypick.vue

```diff
@@ -0,0 +1,146 @@
+            sub-title="Generate destination plate from DART data so it can continue in pipeline partially filled."
+          >
+            <BeckmanCherrypickForm
+              v-slot="{ form, formInvalid }"
+              :action="'create'"
+              :robots="robots"
+            >
+              <b-button
+                variant="success"
+                :disabled="formInvalid"
+                @click="create(form)"
+                >Create Destination Plate</b-button
+              >
+            </BeckmanCherrypickForm>
+          </b-card>
+        </b-tab>
+
+        <b-tab title="Fail">
+          <b-card
+            title="Fail Destination Plate"
+            sub-title="Fail destination plate with a reason."
+          >
+            <BeckmanCherrypickForm
+              v-slot="{ form, formInvalid }"
+              :action="'fail'"
+              :robots="robots"
+              :failure-types="failureTypes"
+            >
+              <b-button
+                variant="danger"
+                :disabled="formInvalid"
+                @click="fail(form)"
+                >Fail Destination Plate</b-button
+              >
+            </BeckmanCherrypickForm>
+          </b-card>
+        </b-tab>
+      </b-tabs>
+    </b-card>
+  </b-container>
+</template>
+
+<script>
+// https://ssg-confluence.internal.sanger.ac.uk/display/PSDPUB/Cherrypicking+Events
+
+import BeckmanCherrypickForm from '@/components/BeckmanCherrypickForm'
+import lighthouse from '@/modules/lighthouse_service'
+import Alert from '@/components/Alert'
+
+export default {
+  components: {
+    BeckmanCherrypickForm,
+    Alert
+  },
+  data() {
+    return {
+      robots: [],
+      failureTypes: []
+    }
+  },
+  async mounted() {
+    await this.getRobots()
+    await this.getFailureTypes()
+  },
+  methods: {
+    async getRobots() {
+      const response = await lighthouse.getRobots()
+
+      if (response.success) {
+        this.robots = response.robots
+      } else {
+        this.robots = []
+        const message = response.errors.join(', ')
+        const type = 'danger'
+        this.showAlert(message, type)
+      }
+    },
+    async getFailureTypes() {
+      const response = await lighthouse.getFailureTypes()
+
+      if (response.success) {
+        this.failureTypes = response.failure_types
+      } else {
+        this.failureTypes = []
+        const message = response.errors.join(', ')
+        const type = 'danger'
+        this.showAlert(message, type)
+      }
+    },
+    async create(form) {
+      let message, type
+
+      const response = await lighthouse.createDestinationPlate(
+        form.username,
+        form.barcode,
+        form.robotSerialNumber
+      )
+      if (response.success) {
+        message = response.response
+        type = 'success'
+      } else {
+        message = response.errors.join(', ')
+        type = 'danger'
+      }
+      this.showAlert(message, type)
+    },
+    async fail(form) {
```

Comment:
Advisory: Maybe you could pass the whole form? rather than passing each as a separate argument.